exasol-jdbc/src/main/scala/com/exasol/spark/util/ExasolOptionsProvider.scala

```diff
@@ -0,0 +1,129 @@
+package com.exasol.spark.util
+
+import java.net.InetAddress
+
+import scala.util.matching.Regex
+
+import com.exasol.errorreporting.ExaError
+import com.exasol.spark.common.ExasolOptions
+import com.exasol.spark.common.Option
+
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+/**
+ * A companion object that creates {@link ExasolOptions}.
+ *
+ * It creates the configuration parameters for Spark Exasol connector.
+ *
+ * These can be user provided when loading or defined in Spark configurations.
+ * For example, user provided:
+ *
+ * {{{
+ *    df = sparkSession
+ *      .read
```

Comment:
This is fine, but I would keep it as is. 

The Spark offers 3 programming APIs (https://spark.apache.org/docs/latest/quick-start.html#self-contained-applications).

For Scala and Python it is without parentheses, for Java it is usual Java method with parenthesis.

In Scala it will even be compile error since the [read method](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala#L740) does not take parameters.